### PR TITLE
[gemini] make warmup ratio AutoPlacementPolicy configurable

### PR DESCRIPTION
**Pull request by feifeibear**

### CODE REVIEW

**Comment by feifeibear** [comment]
```
zero = dict(
    model_config=dict(
        tensor_placement_policy='auto',
        shard_strategy=TensorShardStrategy(),
        reuse_fp16_shard=True,
        **warmup_non_model_data_ratio**=0.9,
    ),
    optimizer_config=dict()
）
```

File: colossalai/zero/sharded_model/sharded_model_v2.py [hunk 1]
```diff
@@ -76,7 +76,9 @@ def __init__(self,
                  fp32_reduce_scatter: bool = False,
                  tensor_placement_policy: str = 'cuda',
                  gradient_predivide_factor: Optional[float] = 1.0,
-                 reuse_fp16_shard: bool = False):
+                 reuse_fp16_shard: bool = False,
+                 *args,
```
Comment (ver217) [inline colossalai/zero/sharded_model/sharded_model_v2.py:80]: It is never used. Remove it?

Comment (feifeibear) [reply colossalai/zero/sharded_model/sharded_model_v2.py:80]: I think it doesn't matter. leaving it here is harmless. Also *args and **kwargs often come together to make the parameters passed in scalable.

Comment (ver217) [reply colossalai/zero/sharded_model/sharded_model_v2.py:80]: `*args` is usaully in front of kwargs like `xxx=xxx`

**Review by ver217 — APPROVED** [review]

**Pull request merged**